navbar.ts

```diff
@@ -0,0 +1,433 @@
+      if (anchorE) parentName.textContent = anchorE.innerHTML;
+      parentName.style.marginRight = "10px";
+
+      const header = document.createElement("div");
+      header.style.padding = "10px";
+      header.style.color = "#ffffff";
+      header.style.display = "flex";
+      header.style.width = "100%";
+
+      header.appendChild(backCloned);
+
+      header.appendChild(parentName);
+      header.appendChild(closeCloned);
+      overlay.appendChild(header);
+
+      const dbox = document.createElement("div");
+
+      const anchor = this.getAnchorElement(anchorName);
+      if (anchor) {
+        const attributes = this.parseAttributes(
+          anchor.getAttribute("attr") || ""
+        );
+        if (attributes.mode == "overlay") {
+          dbox.appendChild(this.overlayNode(attributes.nm));
+        } else if (attributes.mode == "maccordion") {
+          dbox.appendChild(this.accordionCreator(attributes.nm, false));
+        } else {
+          dbox.appendChild(this.accordionCreator(attributes.nm));
+        }
+      }
+      overlay.appendChild(dbox);
+      overlay.style.display = "none";
+      this.overlayContainer[anchorName] = [overlay];
+      if (this.menu) this.menu.appendChild(overlay);
+
+      const clickOutsideListener = (event: MouseEvent) => {
+        if (dbox.contains(event.target as Node)) {
+          document.removeEventListener("click", clickOutsideListener);
+        } else if (!overlay.contains(event.target as Node)) {
+          this.handleClose();
+          document.removeEventListener("click", clickOutsideListener);
+        }
+      };
+
+      if (anchorE) {
+        anchorE.addEventListener("click", (event) => {
+          event.stopPropagation();
+
+          if (this.activeClickOutsideListener) {
+            document.removeEventListener(
+              "click",
+              this.activeClickOutsideListener
+            );
+          }
+
+          document.addEventListener("click", clickOutsideListener);
+          this.activeClickOutsideListener = clickOutsideListener;
+        });
+      }
+
+      closeCloned.addEventListener("click", () => {
+        this.handleClose();
+        if (this.activeClickOutsideListener) {
+          document.removeEventListener(
+            "click",
+            this.activeClickOutsideListener
+          );
+        }
+        this.activeClickOutsideListener = null;
+      });
+    }
+  }
+
+  accordionCreator(
+    parentName: string = "menu",
+    noMultiple: boolean = true
+  ): HTMLElement {
+    const saccordion = document.createElement("aalam-accordion");
+    saccordion.nomultiple = noMultiple;
+    const children = this.parentContainer[parentName];
+    if (parentName == "menu") {
+      const div = document.createElement("div");
+      const divp = document.createElement("div");
+      divp.className = "acc-title";
+      divp.appendChild(this.closeElement);
+      div.appendChild(divp);
+      saccordion.appendChild(div);
+    }

```

Comment:
This thing is getting repeated in the overlay node as well, and we do that only when the parent is the root, so its better have this code in a central place, preferrable at the place where the menu is getting created.